recipe-ts-runtime/src/Oven.ts

```diff
@@ -1,8 +1,23 @@
 import { Cake } from "./Cake";
 import { Recipe } from "./Recipe";
 
+export type Dispatcher = (domain: string | null, payload: string) => string;
+
 export class Oven {
+    private readonly dispatchers: Dispatcher[] = [];
+
     public bake(recipe: Recipe): Cake {
-        return new Cake();
+        let cake = new Cake();
+        for (const segment of recipe.segment()) {
+            const payload = `{"recipe":${segment.recipe.toJson()},"cake":${cake.toJson()}}`;
+            for (const dispatcher of this.dispatchers) {
```

Comment:
Why do we have a list of dispatchers but only record the result of the last one?

Reply:
Because the cake accumulates values as with each dispatch.

Reply:
Seems like a weird API to me then...

Reply:
You're right, this is really weird.... There should only be one dispatcher per domain. I'll update the java one as well in another PR.